editoast/src/views/pathfinding/mod.rs

```diff
@@ -375,12 +375,20 @@ impl Pathfinding {
                 } else {
                     steps_duration.next().unwrap()
                 };
-                let op_name = waypoint
+                let mut op_name = None;
```

Comment:
Rust is designed to avoid `mut` as much as possible. In this case, I see why you did it that way. Here are some ideas to avoid the `mut`.

```rust
let extensions = waypoint
    .id
    .as_ref()
    .map(|op_id| op_map.get(op_id).expect("unexpected OP id"))
    .map(|op| op.extensions);
let (op_name, op_uic) = extensions
    .and_then(|extensions| extensions.identifier.as_ref())
    .map(|ident| (ident.name.as_ref().to_string(), ident.uic))
    .unzip();
let op_ch = extensions
    .and_then(|extensions| extensions.sncf.as_ref())
    .map(|sncf| sncf.ch.to_owned());
```

but you can even avoid the `.unzip()` by flattening all of that.

```rust
let extensions = waypoint
    .id
    .as_ref()
    .map(|op_id| op_map.get(op_id).expect("unexpected OP id"))
    .map(|op| op.extensions);
let op_name = extensions
    .and_then(|extensions| extensions.identifier.as_ref())
    .map(|ident| ident.name.as_ref().to_string());
let op_uic = extensions
    .and_then(|extensions| extensions.identifier.as_ref())
    .map(|ident| ident.uic);
let op_ch = extensions
    .and_then(|extensions| extensions.sncf.as_ref())
    .map(|sncf| sncf.ch.to_owned());
```

Reply:
done with @leovalais